### PR TITLE
[testing_app] added basic placeholder for favorites list

### PR DESCRIPTION
**Pull request by tusharojha**
## Description 

Added a simple placeholder message for the favorites list page.

Related Issue #578

### CODE REVIEW

**Review by RedBrogdon — CHANGES_REQUESTED** [review]
As I mentioned in the associated issue, this app mostly exists to show people how to write tests.

Now that you've added some additional functionality, it's a great opportunity to show how to test that functionality. Consider what kind of test you'd add to this file:

https://github.com/flutter/samples/blob/master/testing_app/test/favorites_test.dart#L31

To make sure the placeholder is correctly displayed when the favorites list is empty.

**Comment by tusharojha** [comment]
I have added a test for checking if the placeholder shows up in the case of the empty favorites list.

Also, While I was going through other tests in the project I found a bit incorrect comment. Although, I haven't changed it since it's not discussed yet.

https://github.com/flutter/samples/blob/master/testing_app/test/models/favorites_test.dart#L14
It says _Generate a random number._ but currently, it's hardcoded to 35. Either we can change the comment or can actually generate a random number.

@RedBrogdon WDYT?

**Comment by abd99** [comment]
> Also, While I was going through other tests in the project I found a bit incorrect comment. Although, I haven't changed it since it's not discussed yet.
https://github.com/flutter/samples/blob/master/testing_app/test/models/favorites_test.dart#L14
It says Generate a random number. but currently, it's hardcoded to 35. Either we can change the comment or can actually generate a random number.

@tusharojha, thanks for mentioning this! I think I missed it while removing the usage of random numbers from the sample. You can remove that comment entirely. I've done the same mistake on [line 25](https://github.com/flutter/samples/blob/0c011574d015777b093fa1413ef70261310342bc/testing_app/test/models/favorites_test.dart#L25) as well. Can you please remove that too?

I should probably have a recheck on the entire sample once this PR gets merged. =)

**Comment by tusharojha** [comment]
> Can you please remove that too?

Yes, of course.

> I should probably have a recheck on the entire sample once this PR gets merged. =)

Although I have gone through the entire code once but definitely a good idea that you also check it once.

File: testing_app/test/favorites_test.dart [hunk 1]
```diff
@@ -28,6 +28,14 @@ void addItems() {
 
 void main() {
   group('Favorites Page Widget Tests', () {
+    testWidgets('Test if Placeholder shows in case of empty list',
+        (tester) async {
+      await tester.pumpWidget(createFavoritesScreen());
+
+      // Verify if Placeholder Text shows up.
```
Comment (abd99) [inline testing_app/test/favorites_test.dart:35]: nit:
```suggestion
      // Verify if the placeholder text shows up.
```

File: testing_app/lib/screens/favorites.dart [hunk 2]
```diff
@@ -16,11 +16,16 @@ class FavoritesPage extends StatelessWidget {
...
-        ),
+        builder: (context, value, child) => value.items.isNotEmpty
+            ? ListView.builder(
+                itemCount: value.items.length,
+                padding: const EdgeInsets.symmetric(vertical: 16),
+                itemBuilder: (context, index) =>
+                    FavoriteItemTile(value.items[index]),
+              )
+            : Center(
+                child: Text('No Favorites Added'),
```
Comment (abd99) [inline testing_app/lib/screens/favorites.dart:27]: nit:
```suggestion
                child: Text('No favorites added.'),
```

File: testing_app/test/favorites_test.dart [hunk 3]
```diff
@@ -28,6 +28,14 @@ void addItems() {
...
+      expect(find.text('No Favorites Added'), findsOneWidget);
```
Comment (abd99) [inline testing_app/test/favorites_test.dart:36]: nit:
```suggestion
      expect(find.text('No favorites added.'), findsOneWidget);
```

**Review by abd99 — CHANGES_REQUESTED** [review]
Thanks for this PR @tusharojha!. I have left a few smalls nits. Other than that, all good.

**Comment by tusharojha** [comment]
Thanks, @abd99!
I am a bit new to these conventions but will take care of these in future commits.

**Review by abd99 — APPROVED** [review]
LGTM!

**Review by RedBrogdon — APPROVED** [review]

**Comment by RedBrogdon** [comment]
Thanks, @tusharojha!

**Pull request merged**